Makefile

```diff
@@ -33,9 +34,20 @@ GOPROXY := https://proxy.golang.org
 endif
 export GOPROXY
 
+# Use GOPRIVATE environment variable if set
+GOPRIVATE := $(shell go env GOPRIVATE)
+export GOPRIVATE
+
 # Active module mode, as we use go modules to manage dependencies
 export GO111MODULE=on
 
+# Base docker images
+
+DOCKERFILE_CONTAINER_IMAGE ?= docker.io/docker/dockerfile:1.4
+DEPLOYMENT_BASE_IMAGE ?= gcr.io/distroless/static
+DEPLOYMENT_BASE_IMAGE_TAG ?= nonroot-${ARCH}
```

Comment:
This changes the default base image from the `latest` tag to `nonroot-amd64` (for example). Was `latest` default a `root` user image? Just making sure this doesn't have unexpected consequences.

Reply:
Yes, seems to Makefile and Dockerfile configurations had different image settings. Please have a look at the docker build log from latest main branch, below, docker pulls **gcr.io/distroless/static:latest** image in Makefile, but docker builds image based on **gcr.io/distroless/static:nonroot-amd64** in Dockerfile and I aligned these configurations. But I we need to use **gcr.io/distroless/static:latest** without $ARCH parameter in tag for image build, please let me know.

```bash
21:15 $ make docker-build
docker pull docker.io/docker/dockerfile:1.4
1.4: Pulling from docker/dockerfile
1328b32c40fc: Already exists 
Digest: sha256:9ba7531bd80fb0a858632727cf7a112fbfd19b17e94c4e84ced81e24ef1a0dbc
Status: Downloaded newer image for docker/dockerfile:1.4
docker.io/docker/dockerfile:1.4
docker pull docker.io/library/golang:1.22.9
1.22.9: Pulling from library/golang
Digest: sha256:89ba45d0421d0a4404249abb3fee78fbc98b01b0bd0bf478d3db9ac35af54495
Status: Downloaded newer image for golang:1.22.9
docker.io/library/golang:1.22.9
docker pull gcr.io/distroless/static:latest
latest: Pulling from distroless/static
Digest: sha256:5c7e2b465ac6a2a4e5f4f7f722ce43b147dabe87cb21ac6c4007ae5178a1fa58
Status: Downloaded newer image for gcr.io/distroless/static:latest
gcr.io/distroless/static:latest
DOCKER_BUILDKIT=1 docker build --build-arg builder_image=docker.io/library/golang:1.22.9 --build-arg goproxy=direct --build-arg ARCH=amd64 --build-arg ldflags="-X 'sigs.k8s.io/cluster-api-addon-provider-helm/version.buildDate=2024-12-30T18:15:54Z' -X 'sigs.k8s.io/cluster-api-addon-provider-helm/version.gitCommit=0cdbb319d3668cbef96e85f6dd24ca5b31f0c65b' -X 'sigs.k8s.io/cluster-api-addon-provider-helm/version.gitTreeState=clean' -X 'sigs.k8s.io/cluster-api-addon-provider-helm/version.gitMajor=0' -X 'sigs.k8s.io/cluster-api-addon-provider-helm/version.gitMinor=2' -X 'sigs.k8s.io/cluster-api-addon-provider-helm/version.gitVersion=v0.2.6-41-0cdbb319d3668c' -X 'sigs.k8s.io/cluster-api-addon-provider-helm/version.gitReleaseCommit=8c307b2fded1e8d9ab996cc3c3dc379942f0f3fb'" . -t localhost:5000/cluster-api-helm-controller-amd64:dev
[+] Building 77.5s (15/17)                                                                                                                                                                                                     docker:default
 => [internal] load build definition from Dockerfile                                                                                                                                                                                     0.0s
 => => transferring dockerfile: 2.61kB                                                                                                                                                                                                   0.0s
 => resolve image config for docker-image://docker.io/docker/dockerfile:1.4                                                                                                                                                              0.0s
 => CACHED docker-image://docker.io/docker/dockerfile:1.4                                                                                                                                                                                0.0s
 => [internal] load .dockerignore                                                                                                                                                                                                        0.0s
 => => transferring context: 228B                                                                                                                                                                                                        0.0s
 => [internal] load metadata for gcr.io/distroless/static:nonroot-amd64                                                                                                                                                                  0.0s
 => [internal] load metadata for docker.io/library/golang:1.22.9                                                                                                                                                                         0.0s
 => [builder 1/8] FROM docker.io/library/golang:1.22.9                                                                                                                                                                                   0.0s
 => [internal] load build context                                                                                                                                                                                                        0.1s
 => => transferring context: 287.99kB                                                                                                                                                                                                    0.0s
 => [stage-1 1/3] FROM gcr.io/distroless/static:nonroot-amd64                                                                                                                                                                            0.0s
 => CACHED [builder 2/8] WORKDIR /workspace                                                                                                                                                                                              0.0s
 => [builder 3/8] COPY go.mod go.mod                                                                                                                                                                                                     0.1s
 => [builder 4/8] COPY go.sum go.sum                                                                                                                                                                                                     0.2s
 => [builder 5/8] RUN --mount=type=cache,target=/go/pkg/mod     go mod download
 ```

Reply:
> Was `latest` default a `root` user image? Just making sure this doesn't have unexpected consequences.

Seems to after #322 fix root image is not needed anymore and provider should work inside the non-privileged container environment without root image.